lambda-esm-msk-filters-sam/README.md

```diff
@@ -0,0 +1,350 @@
+
+1. **base-json.json**
+```
+{
+    "fileNameForValidation": "base-json",
+    "kind": "Event",
+    "apiVersion": "audit.k8s.io/v1",
+    "level": "Request",
+    "requestReceivedTimestamp": "2023-02-06T10:00:00.000000Z",
+    "user": {
+        "username": "eks:pod-identity-mutating-webhook",
+        "groups": [ "system:authenticated" ]
+    },
+    "sourceIPs": [ "10.0.0.1" ],
+    "RBAC": false,
+    "responseStatus":{
+        "metadata":{},
+        "code":200
+    }
+}
+```
+
+2. **code-is-300.json**
+```
+{
+    "fileNameForValidation": "code-is-300",
+    "kind": "Event",
+    "apiVersion": "audit.k8s.io/v1",
+    "level": "Request",
+    "requestReceivedTimestamp": "2023-02-06T10:00:00.000000Z",
+    "user": {
+        "username": "eks:pod-identity-mutating-webhook",
+        "groups": [ "system:authenticated" ]
+    },
+    "sourceIPs": [ "10.0.0.1" ],
+    "RBAC": false,
+    "responseStatus":{
+        "metadata":{},
+        "code":300
+    }
+}
+```
+
+3. **custom-with-region.json**
+```
+{
+    "fileNameForValidation": "custom-with-region",
+    "kind": "Custom",
+    "region": "us-east-1",
+    "apiVersion": "audit.k8s.io/v1",
+    "level": "Request",
+    "requestReceivedTimestamp": "2023-02-06T10:00:00.000000Z",
+    "user": {
+        "username": "eks:pod-identity-mutating-webhook",
+        "groups": [ "system:authenticated" ]
+    },
+    "sourceIPs": [ "10.0.0.1" ],
+    "RBAC": false,
+    "responseStatus":{
+        "metadata":{},
+        "code":200
+    }
+}
+```
+
+4. **rbac-is-set.json**
+```
+{
+    "fileNameForValidation": "rbac-is-set",
+    "kind": "Event",
+    "apiVersion": "audit.k8s.io/v1",
+    "level": "Request",
+    "requestReceivedTimestamp": "2023-02-06T10:00:00.000000Z",
+    "user": {
+        "username": "eks:pod-identity-mutating-webhook",
+        "groups": [ "system:authenticated" ]
+    },
+    "sourceIPs": [ "10.0.0.1" ],
+    "RBAC": true,
+    "responseStatus":{
+        "metadata":{},
+        "code":200
+    }
+}
+```
+
+5. **plain-string.txt**
+```
+OrderNumber:12345
+```
+
+## Testing
+
+We validate the ESM filters by publishing each of the event messages(base-json.jon, code-is-300.json, custom-with-region.json, rbac-is-set.json, plain-string.txt) to the topic that the functions are mapped to.
+
+Login to the client machine in the bastion environment that is configured to publish messages to the topic and run these commands.
+
+```
+#setting broker info
+export BOOTSTRAP_SERVERS_CONFIG="enter cluster's bootstrap brokers"
+#Instructions - https://docs.aws.amazon.com/msk/latest/developerguide/msk-get-bootstrap-brokers.html
+#Sample server config
+#export BOOTSTRAP_SERVERS_CONFIG="b-1.mskclustermsk.abcdef.c14.kafka.us-east-1.amazonaws.com:9092,b-3.mskclustermsk.abcdef.c14.kafka.us-east-1.amazonaws.com:9092,b-2.mskclustermsk.abcdef.c14.kafka.us-east-1.amazonaws.com:9092"
+
+#set topic name
```

Comment:
you are missing the step of creating the topic 

$ bin/kafka-topics.sh --create --topic TopicName --bootstrap-server $BOOTSTRAP_SERVERS_CONFIG

Reply:
I have added this step now. Thanks for reviewing.